### PR TITLE
Lower mutex scope in ThreadStatus

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix performance degradation from #32493.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Not big difference but at least something:

`clickhouse-benchmark --query "SELECT AdvEngineID, count() FROM hits WHERE AdvEngineID != 0 GROUP BY AdvEngineID ORDER BY count() DESC"`

```
Queries executed: 28753.

localhost:9000, queries 28753, QPS: 67.017, RPS: 6701650680.810, MiB/s: 6391.192, result RPS: 1206.297, result MiB/s: 0.010.

0.000%          0.014 sec.
10.000%         0.015 sec.
20.000%         0.015 sec.
30.000%         0.015 sec.
40.000%         0.015 sec.
50.000%         0.015 sec.
60.000%         0.015 sec.
70.000%         0.015 sec.
80.000%         0.015 sec.
90.000%         0.015 sec.
95.000%         0.015 sec.
99.000%         0.016 sec.
99.900%         0.024 sec.
99.990%         0.027 sec.

localhost:9000, queries 25241, QPS: 67.467, RPS: 6746736710.913, MiB/s: 6434.190, result RPS: 1214.413, result MiB/s: 0.010.

0.000%          0.013 sec.
10.000%         0.014 sec.
20.000%         0.014 sec.
30.000%         0.015 sec.
40.000%         0.015 sec.
50.000%         0.015 sec.
60.000%         0.015 sec.
70.000%         0.015 sec.
80.000%         0.015 sec.
90.000%         0.015 sec.
95.000%         0.015 sec.
99.000%         0.016 sec.
99.900%         0.025 sec.
99.990%         0.030 sec.
```

**Comment by alexey-milovidov** [comment]
@novikd Version 21.8:

```
localhost:9000, queries 43632, QPS: 81.722, RPS: 8172170100.558, MiB/s: 7793.589, result RPS: 1470.991, result MiB/s: 0.013.

0.000%          0.011 sec.
10.000%         0.012 sec.
20.000%         0.012 sec.
30.000%         0.012 sec.
40.000%         0.012 sec.
50.000%         0.012 sec.
60.000%         0.012 sec.
70.000%         0.012 sec.
80.000%         0.012 sec.
90.000%         0.012 sec.
95.000%         0.012 sec.
99.000%         0.014 sec.
99.900%         0.069 sec.
99.990%         0.078 sec.
```

File: src/Common/ThreadStatus.cpp [hunk 1]
```diff
@@ -148,28 +148,22 @@ ThreadStatus::ThreadStatus()
...
-            memory_tracker.free(-untracked_memory);
-    }
-    catch (const DB::Exception &)
-    {
-        /// It's a minor tracked memory leak here (not the memory itself but it's counter).
-        /// We've already allocated a little bit more than the limit and cannot track it in the thread memory tracker or its parent.
-        tryLogCurrentException(log);
-    }
+    if (untracked_memory > 0)
+        memory_tracker.allocImpl(untracked_memory, false);
```
Comment (novikd) [inline src/Common/ThreadStatus.cpp:152]:
```suggestion
        memory_tracker.allocNoThrow(untracked_memory);
```

**Review by novikd — APPROVED** [review]

File: programs/server/embedded.xml [hunk 2]
```diff
@@ -12,7 +12,6 @@
 
     <path>./</path>
 
-    <uncompressed_cache_size>8589934592</uncompressed_cache_size>
```
Comment (novikd) [inline programs/server/embedded.xml:15]: This is not related to #32493. Is it an intentional change?

Comment (alexey-milovidov) [reply programs/server/embedded.xml:15]: Yes, just in case.
